narwhals/_pandas_like/dataframe.py

```diff
@@ -276,16 +276,28 @@ def join(
         if isinstance(right_on, str):
             right_on = [right_on]
 
-        return self._from_dataframe(
-            self._dataframe.merge(
-                other._dataframe,
-                left_on=left_on,
-                right_on=right_on,
-                how=how,
-                suffixes=("", "_right"),
-            ),
+        # performing the merge
+        merged_df = self._dataframe.merge(
+            other._dataframe,
+            left_on=left_on,
+            right_on=right_on,
+            how=how,
+            suffixes=("", "_right"),
         )
 
+        start_col = merged_df.columns[0]
+        end_col = [col for col in merged_df.columns if col.endswith("_right")][0]
```

Comment:
does this work even if there's multiple overlapping columns?

Reply:
No, it doesn't. Let me try fix that